packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -127,13 +136,15 @@ export function useShowMoversGestures( {
 		const handleOnFocus = () => {
 			if ( isFocusedWithin() ) {
 				setIsFocused( true );
+				toggleBlockHighlight( clientId, true );
```

Comment:
@draganescu `toggleBlockHighlight` is already called as part of `useShowMoversGestures` here - https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-parent-selector/index.js#L66.

There's probably an issue that prevents `toggleBlockHighight` from being called with `false`, either with the `if` statement here:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-parent-selector/index.js#L63

or here:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-toolbar/utils.js#L79